code/clue.js

```diff
@@ -38,22 +196,90 @@ const randomSelector = array => {
 }
 
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+
+let mystery = {
+  killer: null,
+  weapons: null,
+  room: null
+};
+
 
 // This function will be invoked when you click on the killer card.
+
 const pickKiller = () => {
+
   // This will randomly select a killer from the suspects. And add that to the mystery object.
   mystery.killer = randomSelector(suspects)
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  const killerName = document.getElementById('killerName')
+  const killerOccupation = document.getElementById('killerOccupation')
+  const killerCard = document.getElementById('killerCard')
+  const killerImage = document.getElementById('killerImage');
+
+  killerCard.style.background = mystery.killer.color
+  killerName.innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  killerOccupation.innerHTML = `${mystery.killer.occupation}`;
+  killerImage.src = mystery.killer.image;
+
+  // shuffel favorite weapon function.
+  const shuffleFavouriteWeapon = () => {
+  mystery.favouriteWeapon = randomSelector(weapons);
+  document.getElementById('favouriteWeapon').innerHTML = `Favourite weapon: ${mystery.favouriteWeapon.name}`;
+  }
+  shuffleFavouriteWeapon();
+};
+
+
+// pickWeapon function
+
+const pickWeapon = () => {
+  // This will randomly select a weapon. And add that to the mystery object.
+  mystery.weapons = randomSelector(weapons)
+
+  const weaponCard = document.getElementById('weaponCard')
+  const weaponName = document.getElementById('weaponName')
+
+  weaponCard.style.background = mystery.weapons.color
+  weaponName.innerHTML = `${mystery.weapons.name}`
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+// CREATE FUNCTION pickRoom 
+
+const pickRoom = () => {
+  // This will randomly select a room. And add that to the mystery object.
+  mystery.rooms = randomSelector(rooms)
+
+  const roomName = document.getElementById('roomName')
+  roomName.innerHTML = `${mystery.rooms}`
+}
+
+
+//invoke function onclick
+document.getElementById('killerCard').onclick = pickKiller;
+document.getElementById('weaponCard').onclick = pickWeapon;
+document.getElementById('roomCard').onclick = pickRoom;
```

Comment:
Sometimes it can be helpful to group all of the `onclick` assignments together.  That way you can easily see all of the `onclick` handlers you have on your page.  Can be easier to troubleshoot